hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1023,14 +1023,19 @@ public void update(HoodieRollbackMetadata rollbackMetadata, String instantTime)
 
       validateRollback(commitToRollbackInstantTime, compactionInstant, deltacommitsSinceCompaction);
 
-      // lets apply a delta commit with DT's rb instant(with special suffix) containing following records:
+      // lets apply a delta commit with DT's rb instant containing following records:
       // a. any log files as part of RB commit metadata that was added
       // b. log files added by the commit in DT being rolled back. By rolled back, we mean, a rollback block will be added and does not mean it will be deleted.
       // both above list should only be added to FILES partition.
-
-      String rollbackInstantTime = createRollbackTimestamp(instantTime);
+      if (deltacommitsSinceCompaction.containsInstant(instantTime)) {
+        LOG.info("Rolling back MDT deltacommit for " + instantTime + ", since previous attempt failed");
+        if (!getWriteClient().rollback(instantTime)) {
```

Comment:
so a rollback in DT could trigger 2 rollbacks in MDT?

Reply:
yeah, this was a miss. we can remove this explicit rollbacks since any delta commit to MDT will be handled within commitInternal in HoodieBackedTableMetadataWriter. https://github.com/apache/hudi/blob/759907ebb170cd28b2c8ef8df013be5fd3573fcf/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java#L1101